File: o2o/trunk/java/o2o-web/src/main/java/cn/com/dyninfo/o2o/furniture/web/publish/widget/ShopList.java
/*
 * Copyright (c) 2009-2016 SHENZHEN Eternal Dynasty Technology Co.,Ltd.
 * All rights reserved.
 *
 * This file contains valuable properties of  SHENZHEN Eternal Dynasty
 * Technology Co.,Ltd.,  embodying  substantial  creative efforts  and
 * confidential information, ideas and expressions.    No part of this
 * file may be reproduced or distributed in any form or by  any  means,
 * or stored in a data base or a retrieval system,  without  the prior
 * written permission  of  SHENZHEN Eternal Dynasty Technology Co.,Ltd.
 *
 */

package cn.com.dyninfo.o2o.furniture.web.publish.widget;

import java.util.List;
import java.util.Map;

import javax.annotation.Resource;

import org.springframework.stereotype.Component;

import cn.com.dyninfo.o2o.furniture.util.PageInfo;
import cn.com.dyninfo.o2o.furniture.util.ResponseUtil;

import cn.com.dyninfo.o2o.old.model.AreaInfo;
import cn.com.dyninfo.o2o.old.service.AreaService;
import cn.com.dyninfo.o2o.furniture.web.framework.context.Context;
import cn.com.dyninfo.o2o.furniture.web.framework.facade.Widget;
import cn.com.dyninfo.o2o.furniture.web.publish.model.MerchantType;
import cn.com.dyninfo.o2o.furniture.web.publish.service.MerchantOrderService;
import cn.com.dyninfo.o2o.furniture.web.publish.service.MerchantTypeService;
import cn.com.dyninfo.o2o.furniture.web.publish.service.ShangJiaService;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

@Component("shopList")
public class ShopList extends Widget {

	@Resource
	private MerchantOrderService merchantOrderService;
	@Resource
	private ShangJiaService shangJiaService;
	@Resource
	private AreaService areaService;
	
	@Resource
	private MerchantTypeService merchantTypeService;
	@Override
	public void display(Map pamtr) {
		String action=(String) pamtr.get("action");
		AreaInfo arear=(AreaInfo) this.HttpRequest.getSession().getAttribute(Context.SESSION_AEAR);
		
		if(action.equals("nvrenjie")){//女人街
			PageInfo page=new PageInfo();
			String pageNo=(String) pamtr.get("pageNo");
			String pageSize=(String) pamtr.get("pageSize");
			if(pageNo==null){
				page.setPageNo(1);
			}else{
				page.setPageNo(Integer.parseInt(pageNo));
			}
			page.setPageSize(10);
			String sql="";
			if(arear!=null){
				sql+=" and s.CITY_ID='"+arear.getId()+"' ";
			}
			List list=merchantOrderService.getMerchantPageBywhere("4", sql, page);
			this.putData("shopList", list);
			this.setPageName("nvrenjie.html");
		}else if(action.equals("maptuijian")){//地图店铺推荐
			PageInfo page=new PageInfo();
			String pageNo=(String) pamtr.get("pageNo");
			pamtr.put("pageNo", 1);
			if(pageNo==null){
				page.setPageNo(1);
			}else{
				page.setPageNo(Integer.parseInt(pageNo));
			}
			page.setPageSize(5);
			String sql="";
			if(arear!=null){
				sql+=" and s.CITY_ID='"+arear.getId()+"' ";
			}
			List list=merchantOrderService.getMerchantPageBywhere("3", sql, page);
			if (list.size() == 0) {
				page.setPageNo(1);
				list = merchantOrderService.getMerchantPageBywhere("3", sql, page);
				//System.out.println("由于结果为空，自动将页设为1, 结果数量：" + list.size());
				// 在页面中放入超页标志
				this.putData("pageIndexOut", "1");
			} else {
				this.putData("pageIndexOut", "0");
			}
			
			this.putData("shopList", list);
			this.setPageName("maptuijian.html");
		}else if(action.equals("shopSortName")){
			String sort=getSort();
			MerchantType info=(MerchantType) merchantTypeService.getObjById(sort);
			this.putData("name", info.getName());
			this.putData("sortId", info.getType_id());
			this.setPageName("shopSortName.html");
		}else if(action.equals("shoparea")){
			if(arear!=null){
				AreaInfo city=(AreaInfo) areaService.getObjById(arear.getId());
				this.putData("province", city.getParent());
				this.putData("city", city);
			}
			this.setPageName("area.html");
		}else if(action.equals("tuijian")){
			String sort=getSort();
			PageInfo page=new PageInfo();
			page.setPageNo(1);
			page.setPageSize(5);
			String sql="";
			if(arear!=null){
				sql+=" and s.CITY_ID='"+arear.getId()+"' ";
			}
			if(sort!=null){
				sql+=" and s.TYPE_ID='"+sort+"' ";
			}
			List list=merchantOrderService.getMerchantPageBywhere("2", sql, page);
			this.putData("shopList", list);
			this.setPageName("tuijian.html");
		}else if(action.equals("data")){
			PageInfo page=new PageInfo();
			page.setPageNo(1);
			page.setPageSize(5);
			String sql="";
			String province=(String) pamtr.get("provinceId");
			String city=(String) pamtr.get("cityId");
			String county=(String) pamtr.get("countyId");
			
			if(province!=null){
				sql+=" and s.PROVINCE_ID='"+province+"' ";
			}
			
			/*****************************************************************************
			 * 城市搜索条件（当没有通过传参数时，自动加上当前所在区域的城市ID）
			 ****************************************************************************/
			if(city!=null){
				sql+=" and s.CITY_ID='"+city+"' ";
			} else {
				// 如果是全国，arear必定为空，此时则不加城市ID
				// 如果是其他城市则要加（没有通过下拉框传递参数时才会加上这一句）
				if (arear != null) {
					sql+=" and s.CITY_ID='"+arear.getId()+"' ";
				}
			}
			
			if(county!=null){
				sql+=" and s.COUNT_ID='"+county+"' ";
			}
			String sortId=(String) pamtr.get("sortId");
			if(sortId!=null){
				sql+=" and s.TYPE_ID='"+sortId+"' ";
			}
			List list=merchantOrderService.getMerchantPageBywhere("1", sql, page);
			this.putData("shopList", list);
			this.setPageName("shop.html");
			
		}else if(action.equals("dataTotal")){
			String sql=" and n.state='0' ";
			String province=(String) pamtr.get("provinceId");
			String city=(String) pamtr.get("cityId");
			String county=(String) pamtr.get("countyId");
			
			if(province!=null){
				sql+=" and n.province.id='"+province+"' ";
			}
			
			/*****************************************************************************
			 * 城市搜索条件（当没有通过传参数时，自动加上当前所在区域的城市ID）
			 ****************************************************************************/
			if(city!=null){
				sql+=" and n.city.id='"+city+"' ";
			} else {
				// 如果是全国，arear必定为空，此时则不加城市ID
				// 如果是其他城市则要加（没有通过下拉框传递参数时才会加上这一句）
				if (arear != null) {
					sql+=" and n.city.id='"+arear.getId()+"' ";
				}
			}
			
			if(county!=null){
				sql+=" and n.county.id='"+county+"' ";
			}
			String sortId=(String) pamtr.get("sortId");
			if(sortId!=null){
				sql+=" and n.type.type_id='"+sortId+"' ";
			}
			int count=shangJiaService.getCountByWhere(new StringBuffer(sql));
			this.setFreeMaker(false);
			ResponseUtil.printl(this.HttpResponse, "{\"total\":"+count+"}", "json");
		}else if(action.equals("chain_top")){
			PageInfo page=new PageInfo();
			page.setPageNo(1);
			page.setPageSize(3);
			String sql=" and s.SORT=0 ";
			if(arear!=null){
				sql+=" and s.CITY_ID='"+arear.getId()+"' ";
			}
			List list=merchantOrderService.getMerchantPageBywhere("5", sql, page);
			this.putData("shopList", list);
			this.setPageName("chain_top.html");
		}else if(action.equals("chain_tuijian")){
			PageInfo page=new PageInfo();
			page.setPageNo(1);
			page.setPageSize(5);
			String sql=" and s.SORT=0 ";
			if(arear!=null){
				sql+=" and s.CITY_ID='"+arear.getId()+"' ";
			}
			List list=merchantOrderService.getMerchantPageBywhere("6", sql, page);
			this.putData("shopList", list);
			this.setPageName("chain_tuijian.html");
		}else if(action.equals("chain_data")){
			PageInfo page=new PageInfo();
			String pageNo=(String) pamtr.get("pageNo");
			String pageSize=(String) pamtr.get("pageSize");
			page.setPageNo(Integer.parseInt(pageNo));
			page.setPageSize(Integer.parseInt(pageSize));
			String sql=" and s.SORT=0 ";
			if(arear!=null){
				sql+=" and s.CITY_ID='"+arear.getId()+"' ";
			}
			List list=merchantOrderService.getMerchantPageBywhere("7", sql, page);
			this.putData("shopList", list);
			this.setPageName("chain_data.html");
		}else if(action.equals("chain_count")){
			PageInfo page=new PageInfo();
			page.setPageNo(1);
			page.setPageSize(5);
			String sql=" and n.sort=0 ";
			if(arear!=null){
				sql+=" and n.city.id='"+arear.getId()+"' ";
			}
			int count=shangJiaService.getCountByWhere(new StringBuffer(sql));
			this.setFreeMaker(false);
			ResponseUtil.printl(this.HttpResponse, "{\"total\":"+count+"}", "json");
		}
		
		
		
		
	}
	
	
	private String getSort(){
		String url=this.HttpRequest.getServletPath();
		url=url.substring(url.lastIndexOf("/")+1);
		if(url.indexOf("?")>0)
			url=url.substring(url.indexOf("?"));
		Pattern p=Pattern.compile("([\\d]+)");
		Matcher m=p.matcher(url);
		if(m.find()){
			return m.group(1);
		}
		return null;
	}

}

File: o2o/trunk/java/o2o-web/src/main/webapp/js/shopList.js
/*
 * Copyright (c) 2009-2016 SHENZHEN Eternal Dynasty Technology Co.,Ltd.
 * All rights reserved.
 *  
 * This file contains valuable properties of  SHENZHEN Eternal Dynasty 
 * Technology Co.,Ltd.,  embodying  substantial  creative efforts  and 
 * confidential information, ideas and expressions.    No part of this 
 * file may be reproduced or distributed in any form or by  any  means,
 * or stored in a data base or a retrieval system,  without  the prior 
 * written permission  of  SHENZHEN Eternal Dynasty Technology Co.,Ltd.
 *
 */

$(function(){
	$(".sortBox").click(function(){
		if($(this).attr("class").indexOf("select")>0){
			$(".sortBox.select").find("img").eq(0).attr("src",servicePath+"/img/sp_list_14.gif");
			$(".sortBox.select").next(".splist_kv").css("display","none");
			$(".sortBox.select").removeClass("select")
		}else{
			$(".sortBox.select").find("img").eq(0).attr("src",servicePath+"/img/sp_list_14.gif");
			$(".sortBox.select").next(".splist_kv").css("display","none");
			$(".sortBox.select").removeClass("select")
			$(this).addClass("select").find("img").eq(0).attr("src",servicePath+"/img/sp_list_07.gif");
			$(this).next(".splist_kv").css("display","");
		}
	})

})


page=function(obj,box,widget){
	address("provinceId","cityId","countyId",servicePath+"/widget.html?widget=ArearWidget&action=getArea&dataType=json");
	var _page={
		currentObj:null,
		pageNo:1,
		pageSize:30,
		pageUrl:"",
		total:0,
		totalPage:0,
		htmlBox:null,
		goPage:1,
		widget:"",
		specHeight:10,
		data:"",
		init:function(obj,box,widget){//初始化
			_page.currentObj=obj;
			_page.htmlBox=box;
			_page.widget=widget;
			_page.pageNo=1;
			_page.pageSize=30;
			_page.pageUrl=servicePath+"/widget.html?widget="+widget;
			_page.total=0;
			_page.totalPage=_page.total/_page.pageSize;
			if(_page.totalPage*_page.pageSize<_page.total){
				_page.totalPage++;
			}
			_page.creatHtml();
			_page.addressEvent();
			_page.createPostData();
			
		},
		createPostData:function(){
			var data="";
			if($("#countyId").val().length>0){
				data+="&countyId="+$("#countyId").val();
			}
			if($("#cityId").val().length>0){
				data+="&cityId="+$("#cityId").val();
			}
			if($("#provinceId").val().length>0){
				data+="&provinceId="+$("#provinceId").val();
			}
			data+="&sortId="+$(".sortId").attr("data");
			
			_page.data=data;
			_page.getDataTotal();
		},
		addressEvent:function(){
		
			$("#provinceId").change(function(){
				_page.createPostData();
			});
			$("#cityId").change(function(){
				_page.createPostData();
			});
			$("#countyId").change(function(){
				_page.createPostData();
			});
		},
		getDataTotal:function(){
			$.ajax({
				  type: 'POST',
				  url: _page.pageUrl,
				  data: "action=dataTotal&dataType=json"+_page.data,
				  dataType: "json",
				  success: function(data){
				  		_page.total=data.total;
				  		_page.totalPage=_page.total/_page.pageSize;
				  		if(_page.total%_page.pageSize/_page.pageSize<0.5&&_page.totalPage>0){
				  			_page.totalPage=Math.round(_page.totalPage)+1;
				  		}else{
				  			_page.totalPage=Math.round(_page.totalPage);
				  		}
				  		
				  		_page.goTo();
				  }
				});
		},
		creatHtml:function(){
			var _html='<div class="fenye_shanxia priv"><a href="javascript:;" class="zhu19">上一页</a></div>';
			var page=_page.pageNo;
			if(page>2){
				page-=2;
			}else if(page>1){
				page-=1;
			}
			var endPage=page+5;;
			if(endPage>_page.totalPage){
				endPage=_page.totalPage;
			}
			
			for(var i=page;i<=endPage&&i<=_page.totalPage;i++){
				if(_page.pageNo!=i){
					_html+='<div class="fenye_shu"><a href="javascript:;" class="zhu19 pageBtn">'+i+'</a></div>';
				}else{
					_html+='<div class="fenye_shu">'+i+'</div>';
				}
			}
			if(endPage<_page.totalPage){
				_html+='<div class="fenye_diandian">...</div>';
			}
			_html+='<div class="fenye_shanxia next"><a href="javascript:;" class="zhu19">下一页</a></div>';
			_html+='<div class="fenye_yeshu">共'+_page.totalPage+'页</div>';
			_html+='<div class="fenye_qu">去第</div>';
			_html+='<div class="fenye_tiaok gotoUrl"><input name="" type="text" class="fenye_tiaok_shu"/></div>';
			_html+='<div class="fenye_qu">页</div>';
			_html+='<div class="fenye_go gotoUrlBtn"><a href="javascript:;"><img src="img/sp_list_31.gif" border="0" /></a></div>';
			_page.currentObj.html(_html);
			_page.event();
		},
		event:function(){
			$(".priv").unbind("click").click(function(){
				if(_page.pageNo>1){
					_page.goPage=_page.pageNo-1;
					_page.goTo();
				}
			})
			$(".next").unbind("click").click(function(){
				if(_page.pageNo<_page.totalPage){
					_page.goPage=_page.pageNo+1;
					_page.goTo();
				}
			})
			$(".pageBtn").unbind("click").click(function(){
					_page.goPage=parseInt($(this).html());
					_page.goTo();
			})
			$(".gotoUrlBtn").unbind("click").click(function(){
				var page=$(".gotoUrl input").val();
				var p=/^[0-9]+$/;
				if(!p.test(page)){
					$(this).css("border","#dddddd solid 1px");
				}else{
					if(parseInt(page)>=1&&parseInt(page)<_page.totalPage){
						_page.goPage=parseInt(page);
						_page.goTo();
					}
				}
			});
			
		},
		goTo:function(){
			
			$.ajax({
			  type: 'POST',
			  url: _page.pageUrl,
			  data: "action=data&pageNo="+_page.goPage+"&pageSize="+_page.pageSize+"&dataType=html"+_page.data,
			  dataType: "html",
			  success: function(data){
			  		_page.htmlBox.html(data);
			  		_page.pageNo=_page.goPage;
			  		_page.creatHtml();
			  		_page.goodEvent();
			  }
			});
		},
		goodEvent:function(){
			$(".shop").mouseover(function(){
				$(".sjxsp_kuai_dian").addClass("sjxsp_kuai").removeClass("sjxsp_kuai_dian");
				$(this).addClass("sjxsp_kuai_dian").removeClass("sjxsp_kuai");
			}).mouseout(function(){
				$(this).addClass("sjxsp_kuai").removeClass("sjxsp_kuai_dian");
			})
		}
	}
	_page.init(obj,box,widget);
	
}


function sepecHidden(){
	$(".specBox").css("display","none");
}


